src/schema/sources.ts

```diff
@@ -814,7 +821,26 @@ export const resolvers: IResolvers<any, Context> = {
       info,
     ): Promise<GQLSource> => {
       await ensureSourcePermissions(ctx, id);
-      return getSourceById(ctx, info, id);
+
+      const source = await getSourceById(ctx, info, id);
+      const sourceMember = source.currentMember;
+
+      if (sourceMember) {
+        const referralUrl = new URL(
+          `/squads/${source.handle}/${source.currentMember.referralToken}`,
+          process.env.COMMENTS_PREFIX,
+        );
+
+        if (source.public) {
+          referralUrl.pathname = `/squads/${source.handle}`;
+          referralUrl.searchParams.append('cid', 'squad');
+          referralUrl.searchParams.append('userid', ctx.userId);
+        }
+
+        source.referralUrl = await getShortUrl(referralUrl.toString(), ctx.log);
```

Comment:
This ideally should be part of a graphql field resolver for `referralUrl` or as part of graphorm.
Implementing it only here can result in inconsistencies throughout our api

Reply:
@idoshamun Yeah, understand the concern. 

We would have to optimize `getShortUrl` though to avoid it being run multiple times during single request in that case. I did try in graphql fields and it would run multiple times for each entity inside feed. I used https://vlio20.github.io/utils-decorators/#memoizeAsync for that kind of cases before. Thoughts?

Reply:
We can do an in memory cache in the context or use dataloaders. I think it should solve our problem

Reply:
Yeah, memoizeAsync does the same thing. Never used dataloaders in graphql if you mean this? https://github.com/graphql/dataloader I'll give it a go.

Reply:
Implemented in 8b036510b771a53773d17f6b75d34b4b4ec981da